### PR TITLE
Show differences in status

### LINKED ISSUE #179: Handle status correctly for multiple PVMs
It seems that currently the first VM to stop is causing the whole debugger to stop.

Here I'm running polkavm + pvm-shell.
image

The shell stops at `PC=41` with `HALT` status, but as you can see on the screenshot it seems like all of the PVMs stopped there (i.e. I can't click "Step" any more, "HALT" is displayed as status + there is green higlight at "BRANCH" instruction).

How it should work:
1. The status should display red `?` - because polkavm is still running.
2. There should be no green highlight on BRANCH. There should be green highlight on PC=41 (i.e. where shell stops, but obv that's not displayed). BRANCH should still have blue highlight coming from polkavm.
3. It should still be possible to step over next instruction until polkavm stops too.

### PR DESCRIPTION
**Pull request by wkwiatek**
Fixes #179

### CODE REVIEW

File: src/components/Registers/index.tsx [hunk 1]
```diff
@@ -127,9 +141,19 @@ export const Registers = ({
...
-              <p className="flex-[3]" style={{ color: getStatusColor(currentState.status) }} test-id="program-status">
-                {currentState.status !== undefined ? Status[currentState.status] : null}
-              </p>
+              {currentState.status !== undefined && previousState.status !== undefined && (
+                <ComputedValue
+                  value={currentState.status}
+                  previousValue={previousState.status}
+                  propName="status"
+                  formatter={(value) => (
+                    <span style={{ color: getStatusColor(Number(value) as Status) }} test-id="program-status">
```
Comment (krystian50) [inline src/components/Registers/index.tsx:150]: Maybe try to set Status as value type to prevent casting

File: src/store/workers/workersSlice.ts [hunk 2]
```diff
@@ -299,43 +299,53 @@ export const stepAllWorkers = createAsyncThunk("workers/stepAllWorkers", async (
...
-            instruction: event.data.payload.result,
-          }),
-        );
-
-        // END MOVED FROM initAllWorkers
+  const responses = await Promise.all(
+    state.workers.map((worker) => {
+      return new Promise((resolve: (value: { isFinished: boolean }) => void) => {
+        const messageHandler = (event: MessageEvent<WorkerResponseParams>) => {
+          if (event.data.command === Commands.STEP) {
```
Comment (krystian50) [inline src/store/workers/workersSlice.ts:306]: You can try to refactor and use async post message I added.

**Review by krystian50 — CHANGES_REQUESTED** [review]

**Comment by tomusdrw** [comment]
image

pvm shell showing an empty status, can you check if the value coming from the shell is incorrect? I think it should be within range so should display *something*.

**Review by tomusdrw — APPROVED** [review]
works good! Didn't look at the code.

**Review by krystian50 — APPROVED** [review]

**Pull request merged**